livekit-agents/livekit/agents/voice_assistant/voice_assistant.py

```diff
@@ -359,6 +372,7 @@ def _on_end_of_speech(ev: vad.VADEvent) -> None:
             self.emit("user_stopped_speaking")
             self._deferred_validation.on_human_end_of_speech(ev)
             self._last_end_of_speech_time = time.time()
+            self.set_state("thinking")
```

Comment:
end_of_speech could still happens without necessarily thinking (e.g interruption is disable but the user is speaking).

What you can do is the check if self._playing_speech is present and not done

Reply:
thanks! but do you think this callback is generally the right place to set the `thinking` state or is there a better place to put this?

Reply:
Yes it isn't the right place to put it. VAD can be noisy and after a end of speech. There may be no transcript be (so nothing to answer). I think you can put this inside the _synthesize_agent_reply function.

You can check if the user is still speaking there and only update if not.

Reply:
thanks! I'll leave the debounce delay at `0.0` for now and we can adjust if we deem it to be too noisy?